crates/bevy_ecs/src/world/filtered_resource.rs

```diff
@@ -0,0 +1,576 @@
+            return None;
+        }
+        // SAFETY: We have read access to this resource
+        unsafe { self.world.get_resource_with_ticks(component_id) }.map(
+            |(value, ticks, _caller)| Res {
+                // SAFETY: `component_id` was obtained from the type ID of `R`.
+                value: unsafe { value.deref() },
+                // SAFETY: We have read access to the resource, so no mutable reference can exist.
+                ticks: unsafe { Ticks::from_tick_cells(ticks, self.last_run, self.this_run) },
+                // SAFETY: We have read access to the resource, so no mutable reference can exist.
+                #[cfg(feature = "track_change_detection")]
+                changed_by: unsafe { _caller.deref() },
+            },
+        )
+    }
+
+    /// Gets a pointer to the resource with the given [`ComponentId`] if it exists and the `FilteredResources` has access to it.
+    pub fn get_by_id(&self, component_id: ComponentId) -> Option<Ptr<'w>> {
+        if !self.access.has_resource_read(component_id) {
+            return None;
+        }
+        // SAFETY: We have read access to this resource
+        unsafe { self.world.get_resource_by_id(component_id) }
+    }
+}
+
+impl<'w> From<FilteredResourcesMut<'w>> for FilteredResources<'w> {
+    fn from(resources: FilteredResourcesMut<'w>) -> Self {
+        // SAFETY:
+        // - `FilteredResourcesMut` guarantees exclusive access to all resources in the new `FilteredResources`.
+        unsafe {
+            FilteredResources::new(
+                resources.world,
+                resources.access,
+                resources.last_run,
+                resources.this_run,
+            )
+        }
+    }
+}
+
+impl<'a> From<&'a FilteredResourcesMut<'_>> for FilteredResources<'a> {
+    fn from(resources: &'a FilteredResourcesMut<'_>) -> Self {
+        // SAFETY:
+        // - `FilteredResourcesMut` guarantees exclusive access to all components in the new `FilteredResources`.
+        unsafe {
+            FilteredResources::new(
+                resources.world,
+                resources.access.clone(),
+                resources.last_run,
+                resources.this_run,
+            )
+        }
+    }
+}
+
+impl<'a> From<&'a World> for FilteredResources<'a> {
+    fn from(value: &'a World) -> Self {
+        let mut access = Access::new();
+        access.read_all_resources();
+        let last_run = value.last_change_tick();
+        let this_run = value.read_change_tick();
+        // SAFETY: We have a reference to the entire world, so nothing else can alias with read access to all resources.
+        unsafe {
+            Self::new(
+                value.as_unsafe_world_cell_readonly(),
+                access,
+                last_run,
+                this_run,
+            )
+        }
+    }
+}
+
+impl<'a> From<&'a mut World> for FilteredResources<'a> {
+    fn from(value: &'a mut World) -> Self {
+        Self::from(&*value)
+    }
+}
```

Comment:
I can see wanting to get a `FilteredResources` from a reference to the `World`, but seeing it as a kind of conversion seems weird.

Reply:
My plan here is to make `ReflectResource` take `Into<FilteredResource>` so that it can work with `&World` or `FilteredResource`, similar to how `ReflectComponent` takes `Into<FilteredEntityRef>`.

Reply:
It still seems a bit weird to me, but maybe it won't matter too much. An alternative could be making a trait only for this, though I don't know if it's worth it.